### DISCUSSION TITLE
Problem on minimization of square root value

### DISCUSSION THREAD

**Discussion by Control-HAN** [casadi/casadi#3539] [Q&A] [answered]
Hi, I have a problem when running the following code

    **opti = casadi.Opti();
    
    x = opti.variable();
    opti.subject_to(sqrt(x)>=1);
    opti.minimize(sqrt(x));
    
    opti.solver('ipopt');
    sol = opti.solve();
    sol.value(x)**

**Error in Opti::solve [OptiNode] at .../casadi/core/optistack.cpp:157:
.../casadi/core/optistack_internal.cpp:996: Assertion "return_success(accept_limit)" failed:
Solver failed. You may use opti.debug.value to investigate the latest values of variables. return_status is
'Invalid_Number_Detected'**

I think that 'ipopt' cannot solve the problem on the minimization of square root value.
Two years ago, Joel responded to such problems as
"You probably need to modify the model equations slightly."

If there are someone who can modify the square root value equation, please give me a guidance on this.
Thank you.

**Comment by jgillis** [comment, marked as answer]
Please check the following faq entry: https://github.com/casadi/casadi/wiki/FAQ:-Why-am-I-getting-%22NaN-detected%22in-my-optimization%3F

**Reply by Control-HAN** [reply]
Thank you for your rapid response.

According to FAQ entry, 'ipopt solver' encounters a problem when dealing with sqaure root values like sqrt(x).
So, I changed my problem slightly as
**opti = casadi.Opti();
x = opti.variable();
p = opti.parameter();
opti.set_value(p, 0.0001); 

opti.subject_to(x>=1);
opti.minimize(sqrt(x+p));

opti.solver('ipopt');
sol = opti.solve();
sol.value(x)**

I add the parameter 'p' to avoid the numerical problem.
Is this a similar optimization problem to the original one?
If you are possible, please correct my code.

Thanks again for your help.

**Reply by tinusd** [reply]
If you want to keep your initial problem as is (without p), you need to use another initial solution other than x = 0. Use opti.set_initial(x, 0.5), for example.

**Reply by Control-HAN** [reply]
opti = casadi.Opti();
x = opti.variable();

opti.subject_to(sqrt(x)>=1);
opti.set_initial(x,0.5)
opti.minimize(sqrt(x));

opti.solver('ipopt');
sol = opti.solve();

sol.value(x)

Thank you for your kind response^^
